front-end-tools/src/components/InitComponent.tsx

```diff
@@ -0,0 +1,643 @@
+                    });
+            }, REFRESH_INTERVAL.asMilliseconds());
+        }
+    }, [connection, account, client, txHash]);
+
+    useEffect(() => {
+        if (connection && client && account && moduleReference) {
+            if (moduleReference.length === 64) {
+                client
+                    .getModuleSource(new ModuleReference(moduleReference))
+                    .then((value) => {
+                        if (value === undefined) {
+                            setIsModuleReferenceAlreadyDeployedStep2(false);
+                        } else {
+                            setIsModuleReferenceAlreadyDeployedStep2(true);
+                        }
+                    })
+                    .catch(() => {
+                        setIsModuleReferenceAlreadyDeployedStep2(false);
+                    });
+            }
+        }
+    }, [connection, account, client, moduleReference]);
+
+    useEffect(() => {
+        if (
+            moduleReference !== undefined &&
+            moduleReferenceCalculated !== undefined &&
+            moduleReferenceCalculated !== moduleReference
+        ) {
+            setShouldWarnDifferenceModuleReferences(true);
+        } else {
+            setShouldWarnDifferenceModuleReferences(false);
+        }
+    }, [moduleReference, moduleReferenceCalculated]);
+
+    useEffect(() => {
+        if (inputParameterTemplate !== undefined && hasInputParameter === false) {
+            setShouldWarnInputParameterInSchemaIgnored(true);
+        } else {
+            setShouldWarnInputParameterInSchemaIgnored(false);
+        }
+    }, [inputParameterTemplate, hasInputParameter]);
+
+    useEffect(() => {
+        setSchemaError(undefined);
+        setInputParameterTemplate(undefined);
+
+        let initTemplate;
+
+        try {
+            if (smartContractName === undefined) {
+                throw new Error('Set smart contract name');
+            }
+
+            let schema = '';
+
+            const schemaFromModule = useModuleReferenceFromStep1
+                ? embeddedModuleSchemaBase64
+                : uploadedModuleSchemaBase64;
+
+            if (schemaFromModule !== undefined) {
+                schema = schemaFromModule;
+            }
+
+            const inputParamterTypeSchemaBuffer = getInitContractParameterSchema(
+                toBuffer(schema, 'base64'),
+                smartContractName,
+                2
+            );
+
+            initTemplate = displayTypeSchemaTemplate(inputParamterTypeSchemaBuffer);
+
+            setInputParameterTemplate(initTemplate);
+        } catch (e) {
+            if (useModuleReferenceFromStep1) {
+                setSchemaError(
+                    `Could not get embedded schema from the uploaded module. Uncheck "Use Module from Step 1" checkbox to manually upload a schema or uncheck "Has Input Paramter" checkbox if this entrypoint has no input parameter.  Original error: ${e}`
```

Comment:
Would it not make sense to instead just offer to accept a raw parameter (in the form of a hex string)?

This allows somebody to use a pre-serialized string, or to deploy a new instance from parameters of a different init transaction without caring about the schema.

Reply:
This is very useful for experienced developers also in the context of e.g. using proxy-implementation contracts where the schema of a contract might not include all the options on how to interact with the contract.

Since this feature needs some substantial changes, I added it to the top of the requested feature list here:
https://github.com/Concordium/concordium-smart-contract-tools/issues/131